.github/workflows/build.yml

```diff
@@ -0,0 +1,269 @@
+        with:
+          path-type: inherit
+          install: m4
+
+      - name: Checkout repository
+        if: ${{ !inputs.fake }}
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.channel }}
+
+      - name: Cache Rust dependencies
+        if: ${{ !inputs.fake }}
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~\.cargo\bin
+            ~\.cargo\registry\index
+            ~\.cargo\registry\cache
+            ~\.cargo\git\db
+            ~\AppData\Local\gmp-mpfr-sys
+            rust\target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('rust/Cargo.toml') }}
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Create Rust build for windows
+        if: ${{ !inputs.fake }}
+        run: bash tools/rust_build.sh -i -r -t -f "$FEATURES"
+
+      - name: Create Rust build for windows (FAKE)
+        if: ${{ inputs.fake }}
+        run: |
+          mkdir -p rust/target/release/
+          echo windows >rust/target/release/opendp.dll
+
+      - name: Upload .dll
+        uses: actions/upload-artifact@v3
+        with:
+          name: libs
+          # Unnecessary '*' in path will cause artifact to include parent directories starting at that element.
+          path: rust/target*/release/opendp.dll
+
+  libs-macos:
+    needs: credential-check
+    runs-on: macos-11
+    steps:
+      - name: Checkout repository
+        if: ${{ !inputs.fake }}
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.channel }}
+
+      - name: Cache Rust dependencies
+        if: ${{ !inputs.fake }}
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin
+            ~/.cargo/registry/index
+            ~/.cargo/registry/cache
+            ~/.cargo/git/db
+            ~/Library/Caches/gmp-mpfr-sys
+            rust/target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('rust/Cargo.toml') }}
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Create Rust build for macos x86_64
+        if: ${{ !inputs.fake }}
+        run: bash tools/rust_build.sh -i -r -t -g x86_64-apple-darwin -f "$FEATURES"
+        env:
+          CC: clang -target x86_64-apple-darwin
+
+      - name: Create Rust build for macos aarch64 (no tests)
+        if: ${{ !inputs.fake }}
+        run: bash tools/rust_build.sh -i -r -g aarch64-apple-darwin -f "$FEATURES"
+        env:
+          CC: clang -target aarch64-apple-darwin
+
+      - name: Create Rust build for macos universal
+        # https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary#Update-the-Architecture-List-of-Custom-Makefiles
+        if: ${{ !inputs.fake }}
+        run: |
+          mkdir -p rust/target/release
+          lipo \
+              rust/target/x86_64-apple-darwin/release/libopendp.dylib \
+              rust/target/aarch64-apple-darwin/release/libopendp.dylib \
+              -output rust/target/release/libopendp.dylib -create
```

Comment:
I was on the fence about whether to move macOS to use separate libs. I left it because it was less churn, but now I realize that it could make builds a little quicker, because these could be done in parallel. I'll switch it.